perception/autoware_tensorrt_bevdet/src/gatherbev_plugin.cu

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "autoware/tensorrt_bevdet/common.hpp"
+#include "autoware/tensorrt_bevdet/gatherbev_plugin.hpp"
+
+#include <cuda_fp16.h>
+#include <cuda_runtime_api.h>
+
+// input[0] == adj_feat   b x 8 x 80 x 128 x 128
+// input[1] == curr_feat  b x 80 x 128 x 128
+// input[2] == flag       b x 1
+// out[0]                 b x (8+1)*80 x 128 x 128
+template <typename T>
+__global__ void copy_feat_kernel(
+  int nthreads,  // b * (adj_num + 1) * map_size
+  int adj_num, int channel, int map_size, const T * adj_feats, const T * curr_feat,
+  const int * flag, T * out_feats)
+{
+  CUDA_1D_KERNEL_LOOP(idx, nthreads)
+  {
+    int b = idx / ((adj_num + 1) * map_size);
+    int n = (idx / map_size) % (adj_num + 1);
+    int m = idx % map_size;
+
+    int start = b * (adj_num + 1) * channel * map_size + n * channel * map_size + m;
+    int end = start + channel * map_size;
+    for (int i = start, c = 0; i < end; i += map_size, c++) {
+      if (flag[b] == 0 || n == 0) {
+        out_feats[i] = curr_feat[b * channel * map_size + c * map_size + m];
+      } else {
+        out_feats[i] = adj_feats[i - channel * map_size];
+      }
+    }
+  }
+}
+
+namespace nvinfer1
```

Comment:
Consider wrapping the entire `GatherBEVPlugin` class within a namespace.  This will help prevent naming conflicts with other parts of the project and improve code organization. For example:

```cpp
namespace autoware_tensorrt_bevdet {

class GatherBEVPlugin {
  // ... class definition ...
};

}  // namespace autoware_tensorrt_bevdet
```

This should be applied consistently throughout the file for all classes and functions.